src/config/license.rs

```
@@ -161,11 +165,15 @@ impl Config {
         };
 
         let (end_year, start_year) = if self.use_dynamic_year_ranges {
+            let now_date = Local::now().format("%a %b %d %T %Y %z").to_string();
             let dates = get_git_dates_for_file(filename);
             let (last_updated_date, created_date) = match &dates[..] {
                 [first_date, .., last_date] => (first_date, last_date),
                 [first_date] => (first_date, first_date),
-                _ => panic!("Did not get any dates from git!"),
+                _ => {
+                    debug!("Did not get any dates from git for file: {}", filename);
+                    (&now_date, &now_date)
```

I think this violates the principle of least surprise.

As a user if I've opted into this feature I expect everything to have a git date and use those. Just throwing up some default of the current date silently is a bad UX.

So I really have a question and a change if the answer to the question is yes:

1. Is using the current date really the best we can do here? Why is it better than failing loudly and clearly? (For example what about ignoring the file)
2. If we are going to return the current date because there's no git info then I don't think a debug log suffices and we should loudly let the user know that we're doing something that isn't obvious or expected.

---

My intended use case here is to run Licensure to apply a license header before committing new files (possibly even as a git pre-commit hook). So, failing or ignoring would be annoying in that case, as would requiring them to be in Git already.

"Is using the current date really the best we can do here?"
Using the current date is already the default elsewhere in Licensure (making it obvious and expected to me at least) and I'm not sure what would be better than that in this case, though I'm happy to hear ideas. I'm not super opposed to slightly louder logging about this, but I'm struggling to think of cases where this goes bad.

Let me know if this makes a bit more sense now and/or what log level you think is appropriate.

---

Ok I think that's a valid use case and makes sense to me, I'm not actually using this feature myself so it hadn't occurred to me.

I would like to change how this whole feature works a bit based on this conversation but I will do so after I merge your branch. So this is fine as-is

---

Not sure what you're thinking as far as changes, but I am certainly interested in the feature. Not sure if you're aware of this crate, but it might be helpful: https://github.com/BurntSushi/ripgrep/tree/master/crates/ignore